src/big/big_float.cr

```diff
@@ -51,16 +83,26 @@ struct BigFloat < Float
     LibGMP.mpf_cmp(self, other)
   end
 
-  def <=>(other : Float)
+  def <=>(other : BigInt)
+    LibGMP.mpf_cmp_z(self, other)
+  end
+
+  def <=>(other : Float32 | Float64)
     LibGMP.mpf_cmp_d(self, other.to_f64)
   end
 
-  def <=>(other : Int::Signed)
-    LibGMP.mpf_cmp_si(self, other.to_i64)
+  def <=>(other : Int)
```

Comment:
Might as well leave off the type restriction here since the `else` clause is the same as the `Number` overload.

Reply:
Crystal overloading has always worked for me. Please don't avoid language features for bugs which you can't reproduce. Crystal's overloading is one of it's nicest features, being capable of being dispatched dynamically at runtime, and always ending up at the most specific overload.

Reply:
However, I do think the `other.is_a?(Int8 | Int16 | Int32) || (LibGMP::Long == Int64 && other.is_a?(Int64))` condition is the cleanest way to express what is required.

Reply:
just reviewed <=>, but have not found advantage of separating `Int::Signed` and `Int::Unsigned` overloads. Anyway there will be `else` branch for 32bit arch.

Reply:
What I mean is
```cr
def <=>(other : Number)
  if other.is_a?(Int8 | Int16 | Int32) || (LibGMP::Long == Int64 && other.is_a?(Int64))
    LibGMP.mpf_cmp_si(self, other)
  elsif other.is_a?(UInt8 | UInt16 | UInt32) || (LibGMP::ULong == UInt64 && other.is_a?(UInt64))
    LibGMP.mpf_cmp_ui(self, other)
  else
    LibGMP.mpf_cmp(self, other.to_big_f)
  end
end
```

and delete 
```cr
def <=>(other : Number)
  LibGMP.mpf_cmp(self, other.to_big_f)
end
```
entirely.

Reply:
Thanks :-)